app/src/main/java/com/hynekbraun/openmeteoweather/presentation/mainscreen/WeatherViewModel.kt

```diff
@@ -0,0 +1,106 @@
+package com.hynekbraun.openmeteoweather.presentation.mainscreen
+
+import android.location.Location
+import android.util.Log
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.hynekbraun.openmeteoweather.domain.CurrentLocationManager
+import com.hynekbraun.openmeteoweather.domain.LocationError
+import com.hynekbraun.openmeteoweather.domain.WeatherFetchError
+import com.hynekbraun.openmeteoweather.domain.WeatherRepository
+import com.hynekbraun.openmeteoweather.domain.mapper.toCurrentData
+import com.hynekbraun.openmeteoweather.domain.mapper.toCurrentHourlyForecastData
+import com.hynekbraun.openmeteoweather.domain.mapper.toDailyForecastData
+import com.hynekbraun.openmeteoweather.domain.util.Resource
+import com.hynekbraun.openmeteoweather.presentation.mainscreen.util.*
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class WeatherViewModel @Inject constructor(
+    private val repository: WeatherRepository,
+    private val currentLocation: CurrentLocationManager
+) : ViewModel() {
+
```

Comment:
Regarding the whole view model logic, would it be better to have one function observing the database (observer) and one fetching new data (fetcher)? So I would call observer in init function and call fetcher in ON_START in Main Activity and in repository it would just update db? In my opinion, it feels like it does not matter that much here and is a matter of choice. The only difference I see is that the way I have it you just fire the event, display data and thats it, nothing more is happening. But if you observe it, it is heavier, isn't it? Is observing something to think about performance-wise?